test/sequential/test-net-connect-local-error.js

```diff
@@ -3,28 +3,21 @@ const common = require('../common');
 const assert = require('assert');
 const net = require('net');
 
+const expectedErrorCodes = ['ECONNREFUSED', 'EADDRINUSE'];
 const client = net.connect({
-  port: common.PORT + 1,
-  localPort: common.PORT,
+  port: common.PORT,
+  localPort: common.PORT + 1,
   localAddress: common.localhostIPv4
 });
 
 client.on('error', common.mustCall(function onError(err) {
+  assert.ok(expectedErrorCodes.includes(err.code));
   assert.strictEqual(err.syscall, 'connect');
-  assert.strictEqual(err.code, 'ECONNREFUSED');
-  assert.strictEqual(
-    err.localPort,
-    common.PORT,
-    `${err.localPort} !== ${common.PORT} in ${err}`
-  );
-  assert.strictEqual(
-    err.localAddress,
-    common.localhostIPv4,
-    `${err.localAddress} !== ${common.localhostIPv4} in ${err}`
-  );
+  assert.strictEqual(err.localPort, common.PORT + 1);
+  assert.strictEqual(err.localAddress, common.localhostIPv4);
   assert.strictEqual(
     err.message,
-    `connect ECONNREFUSED ${err.address}:${err.port} ` +
+    `connect ${err.code} ${err.address}:${err.port} ` +
```

Comment:
AFACT they come from the same mechanism.
I tried to stress this, but didn't want to hog the freeBSD machine so only run 50 cycles.
Last time we saw 1 in a 100 give the `EADDRINUSE`.
So worst case we have a flake and some good comments and bread crumbs to dig further.